app/vep/models/pipeline_model.py

```diff
@@ -1,21 +1,65 @@
 from typing import Optional, List, Literal
 from pydantic import BaseModel, validator
+from core.config import NF_COMPUTE_ENV_ID, VEP_CONFIG_INI_PATH
+
+import tempfile 
+
 class VEPConfigParams(BaseModel):
     vcf: str
     vep_config: str
+
 class LaunchParams(BaseModel):
-    computeEnvId: str = "17wuWKrWOZW5JrLoGGwPdD"
+    computeEnvId: str = NF_COMPUTE_ENV_ID
     pipeline: str
     workDir: str
     revision: str = "main"
-    pullLatest: bool = "true"
+    pullLatest: bool = True
     configProfiles: List[str] = ["slurmnew"]
     paramsText: VEPConfigParams
     preRunScript: str = "module load nextflow"
     postRunScript: str = ""
-    stubRun: bool = "false"
+    stubRun: bool = False
     labelIds: List[str]
     headJobCpus: int = 1
     headJobMemoryMb: int = 4
+
 class PipelineParams(BaseModel):
     launch: LaunchParams
+
+class ConfigIniParams(BaseModel):
+  cache: int = 1
+  dir_cache: str = "/nfs/production/flicek/ensembl/variation/data/VEP/tabixconverted/"
+  species: str = "homo_sapiens"
+  assembly: str = 'GRCh38'
+  cache_version: int = 110
+  offline: int = 1
+  force_overwrite: int = 1
+  symbol: int = 1
+  biotype: int = 1
+  transcript_version: int = 1
+  canonical: int = 1
+  ini_file: str
+
+  # Creates config ini file
+  def create_config_ini_file(self, symbol: bool = False, biotype: bool = False):
```

Comment:
How is this method going to be called?

If the caller is going to run:

```
config_ini = ConfigIniParams(symbol=True, biotype=True)
ini_file = config_ini.create_config_ini_file()
```

then the values of symbol and biotype will be passed into the class instance during its creation; and there will not be any need to pass any arguments into the `create_config_ini_file` method, which would read the values of symbol and biotype off the class instance fields.

Reply:
At the moment that method doesn't return anything, instead it populates the class member variable `ini_file`
Regarding your suggestion @azangru I am not sure how ConfigIniParams work without BaseModel. @Mehrnaz-Charkhchi do you know?